service/service.go

```diff
@@ -121,10 +128,32 @@ func (svc *Service) Close(ctx context.Context) error {
 			svc.healthCheck.Stop()
 		}
 
-		// stop any incoming requests before closing any outbound connections
-		shutDownErr = svc.server.Shutdown(ctx)
+		// If kafka consumer exists, close it.
```

Comment:
I think the order should be:

1 - stop healthcheck (so the subscription-based consumption will finish)
2 - consumer StopAndWait (so it finishes any in-progress message and then stops consuming)
3A - close server
3B - close producer
3C - close consumer
(I'd say 3A, 3B, 3C are interchangeble or could even be run in parallel)

Reply:
Ok, I don't think the subscription/stop-and-wait pattern has been added to this service yet, will come as part of the next ticket.

Reply:
ok yes we can change this later if needed